src/PSServiceBus/Helpers/SbReceiver.cs

```diff
@@ -134,11 +134,29 @@ private IList<SbMessage> BuildMessageList(IList<Message> Messages)
                 result.Add(new SbMessage
                 {
                     MessageId = message.MessageId,
-                    MessageBody = ConvertMessageBodyToString(message.Body)
+                    MessageBody = ConvertMessageBodyToString(message.Body),
+                    SystemProperties = ConvertSystemPropertiesToIDictonary(message.SystemProperties),
+                    UserProperties = message.UserProperties
                 });
             }
 
             return result;
         }
+
+        private IDictionary<string, object> ConvertSystemPropertiesToIDictonary(Message.SystemPropertiesCollection systemProperties)
```

Comment:
Would you say it is necessary to convert the `Message.SystemProperties` to an `IDictionary`?  If we _don't_ convert the system properties will still be accessible to the user like so:

```
PS>$msg = Receive-SbMessage -QueueName $queueName -NumberOfMessagesToReceive 1
PS>$msg.SystemProperties
```

This would make the code cleaner and would also prevent a change from being required if MSFT add/remove members from `Message.SystemPropertiesCollection`.

Reply:
I had the same idea at first, but when doing a Get-Member on the message, the preview isn't as informing.

The UserProperties is a `IDictionary`, so from an end user perspective, I wanted to give them the same look and feel. But yes, it will require us to keep things updated. Or write the parser logic to use reflection and iterate over the SystemPropertiesCollection.

My personal believe is that we should aid the end user as much as possible and make things behave the same way. Having to set of properties, but with different types aren't going to help the end user.

Reply:
Ah yeh that makes sense, I'm definitely on board with keeping `UserProperties` and `SystemProperties` consistent for the user.  

I propose we proceed as you have suggested - keeping the conversion static for now - and address this if it presents a problem later on (i.e. if MSFT make changes).

Reply:
I'm glad that we are on the same page and thanks for taking the time to review things and provide feedback!